docarray/array/mixins/io/pushpull.py

```diff
@@ -104,43 +127,55 @@ def gen():
             yield _tail
 
         res = requests.post(
-            f'{_get_cloud_api()}/v2/rpc/da.push', data=gen(), headers=headers
+            f'{_get_cloud_api()}/v2/rpc/artifact.upload', data=gen(), headers=headers
         )
+        json_res = res.json()
 
         if res.status_code != 200:
-            json_res = res.json()
             raise RuntimeError(
                 json_res.get('message', 'Failed to push DocumentArray to Jina Cloud'),
                 f'Status code: {res.status_code}',
             )
 
+        return json_res.get('data')
+
     @classmethod
     def pull(
         cls: Type['T'],
-        token: str,
+        name: str,
         show_progress: bool = False,
         local_cache: bool = False,
         *args,
         **kwargs,
     ) -> 'T':
         """Pulling a :class:`DocumentArray` from Jina Cloud Service to local.
 
-        :param token: the upload token set during :meth:`.push`
+        :param name: the upload name set during :meth:`.push`
         :param show_progress: if to show a progress bar on pulling
         :param local_cache: store the downloaded DocumentArray to local folder
         :return: a :class:`DocumentArray` object
         """
 
         import requests
 
-        url = f'{_get_cloud_api()}/v2/rpc/da.pull?token={token}'
-        response = requests.get(url)
+        headers = {}
+
+        auth_token = _get_hub_config().get('auth_token')
+        if auth_token:
+            headers['Authorization'] = f'token {auth_token}'
+
+        url = f'{_get_cloud_api()}/v2/rpc/artifact.getDownloadUrl?name={name}'
+        response = requests.get(url, headers=headers)
 
         if response.ok:
             url = response.json()['data']['download']
         else:
-            raise FileNotFoundError(
-                f'can not find `{token}` DocumentArray on the Cloud. Misspelled?'
+            json_res = response.json()
+            raise Exception(
```

Comment:
```suggestion
            raise ObjectNotFoundError(
```
and let's create a docarray/exceptions.py module and include such an error

Reply:
addressed